test/controllers/tasks/active.officers.controller.unit.ts

```diff
@@ -1,77 +1,100 @@
+import { getActiveOfficerDetailsData } from "../../../src/services/active.officer.details.service";
 
 jest.mock("../../../src/middleware/company.authentication.middleware");
 jest.mock("../../../src/services/company.officers.service");
 jest.mock("../../../src/utils/feature.flag");
 
 const mockCompanyAuthenticationMiddleware = companyAuthenticationMiddleware as jest.Mock;
-const mockGetCompanyOfficers = getCompanyOfficers as jest.Mock;
-const mockIsActiveFeature = isActiveFeature as jest.Mock;
 mockCompanyAuthenticationMiddleware.mockImplementation((req, res, next) => next());
+const mockGetActiveOfficerDetails = getActiveOfficerDetailsData as jest.Mock;
 
 const COMPANY_NUMBER = "12345678";
-const EXPECTED_TEXT = "Review the directors";
+const PAGE_HEADING = "Check the director's details";
 const EXPECTED_ERROR_TEXT = "Sorry, the service is unavailable";
-const url = ACTIVE_OFFICERS_PATH.replace(":companyNumber", COMPANY_NUMBER);
+const ACTIVE_OFFICER_DETAILS_URL = ACTIVE_OFFICERS_PATH.replace(`:${urlParams.PARAM_COMPANY_NUMBER}`, COMPANY_NUMBER);
 const TASK_LIST_URL = TASK_LIST_PATH.replace(`:${urlParams.PARAM_COMPANY_NUMBER}`, COMPANY_NUMBER);
 
 describe("Active officers controller tests", () => {
 
   beforeEach(() => {
     mocks.mockAuthenticationMiddleware.mockClear();
+    mocks.mockServiceAvailabilityMiddleware.mockClear();
+    mocks.mockSessionMiddleware.mockClear();
+    mockGetActiveOfficerDetails.mockClear();
   });
 
-  it("Should return active officers page", async () => {
-    mockIsActiveFeature.mockReturnValueOnce(true);
-    mockGetCompanyOfficers.mockResolvedValueOnce(validCompanyOfficers);
-    const response = await request(app).get(url);
-    expect(response.text).toContain(EXPECTED_TEXT);
-  });
+  describe("get tests", () => {
 
-  it("Should return error page if active officers cannot be retired", async () => {
-    const message = "Can't connect";
-    mockGetCompanyOfficers.mockRejectedValueOnce(new Error(message));
-    mockIsActiveFeature.mockReturnValueOnce(true);
-    const response = await request(app).get(url);
-    expect(response.text).toContain(EXPECTED_ERROR_TEXT);
-  });
+    it("Should navigate to director's details page", async () => {
+      mockGetActiveOfficerDetails.mockResolvedValueOnce(mockActiveOfficerDetails);
+      const response = await request(app).get(ACTIVE_OFFICER_DETAILS_URL);
 
-  it("Should skip officer check and return active officers page if feature flag is off", async () => {
-    mockIsActiveFeature.mockReturnValueOnce(false);
-    const response = await request(app).get(url);
-    expect(response.text).toContain(EXPECTED_TEXT);
-  });
+      expect(response.text).toContain(PAGE_HEADING);
+      expect(response.text).toContain("Is the director still active?");
+      expect(response.text).toContain(mockActiveOfficerDetails.foreName1);
+      expect(response.text).toContain(mockActiveOfficerDetails.dateOfBirth);
```

Comment:
could you add an expect to check forename2 appears in the response.text and add another test to run with no forename2 (undefined). For the second test you could just set mockActiveOfficerDetails.forname2 to undefined and set it back to it's original value at the end of that test. (you'll have to store the value in a const so you can set forename2 to that value at end of test).